pydoctor/model.py

```diff
@@ -674,7 +678,37 @@ def __init__(self, options: Optional['Options'] = None):
         # We use the fullName of the objets as the dict key in order to bind a full name to a privacy, not an object to a privacy.
         # this way, we are sure the objects' privacy stay true even if we reparent them manually.
         self._privacyClassCache: Dict[str, PrivacyClass] = {}
+        
+        # workaround cyclic import issue
+        from pydoctor import extensions
+
+        # Initialize the extension system
+        self.factory = factory.Factory()
+        self.astbuilder_visitors: Set[Type['astutils.NodeVisitorExt']] = set()
```

Comment:
sets are probably not the best interface for these, we might introduce priorities for each extension visitors or post processors.